File: test/db_test.py
import os
import sys
from dotenv import load_dotenv
from sqlalchemy import create_engine, text

def test_db_connection():
    # Load environment variables from the backend/.env file
    env_path = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'backend', '.env'))
    if not os.path.exists(env_path):
        print(f"ERROR: .env file not found at {env_path}")
        return

    load_dotenv(env_path)
    
    db_url = os.getenv("DATABASE_URL")
    if not db_url:
        print("ERROR: DATABASE_URL not found in environment variables.")
        return

    print("Attempting to connect to database using URL from .env...")
    
    try:
        # Create SQLAlchemy engine
        engine = create_engine(db_url)
        
        # Try to connect and run a simple query
        with engine.connect() as conn:
            version = conn.execute(text("SELECT version();")).scalar()
            print("\n✅ Successfully connected to PostgreSQL!")
            print(f"Database version: {version}")
            
            # Let's check if our tables exist
            print("\nChecking for SiteSync tables in 'public' schema...")
            tables_result = conn.execute(text("SELECT tablename FROM pg_tables WHERE schemaname = 'public';"))
            tables = [row[0] for row in tables_result]
            
            if tables:
                print(f"Found {len(tables)} tables:")
                for table in sorted(tables):
                    print(f"  - {table}")
            else:
                print("No tables found in the public schema. Did you run the schema.sql script?")
                
    except Exception as e:
        print("\n❌ FAILED to connect to the database.")
        print(f"Error details:\n{e}")

if __name__ == "__main__":
    test_db_connection()

File: test/db_test.js
const fs = require('fs');
const path = require('path');

// Manually parse frontend/.env to avoid needing dotenv package
const envPath = path.join(__dirname, '../frontend/.env');
const envContent = fs.readFileSync(envPath, 'utf8');
const env = {};
envContent.split('\n').forEach(line => {
  line = line.trim();
  if (!line || line.startsWith('#')) return;
  const parts = line.split('=');
  const key = parts[0].trim();
  const val = parts.slice(1).join('=').trim();
  env[key] = val;
});

const supabaseUrl = env.NEXT_PUBLIC_SUPABASE_URL;
const supabaseKey = env.NEXT_PUBLIC_SUPABASE_ANON_KEY;

if (!supabaseUrl || !supabaseKey) {
  console.error("Missing Supabase credentials in frontend/.env");
  process.exit(1);
}

// We will use the supabase-js client already installed in the frontend
const { createClient } = require('../frontend/node_modules/@supabase/supabase-js');

const supabase = createClient(supabaseUrl, supabaseKey);

async function testConnection() {
  console.log(`Attempting to connect to Supabase at: ${supabaseUrl}`);
  
  // Test by querying the companies table we created in the schema
  const { data, error } = await supabase.from('companies').select('*').limit(3);
  
  if (error) {
    console.error("\n❌ FAILED to query Supabase.");
    console.error(error.message);
    if (error.message.includes('relation "public.companies" does not exist')) {
        console.error("The tables don't exist yet. Make sure to run schema.sql in Supabase SQL editor!");
    }
  } else {
    console.log("\n✅ Successfully connected to Supabase!");
    console.log(`Found ${data.length} records in 'companies' table.`);
    if (data.length > 0) {
      console.log(data);
    } else {
      console.log("(Table is empty, which is normal for a fresh database)");
    }
  }
}

testConnection();
